interp/interp_test.go

```diff
@@ -1038,3 +1041,34 @@ func TestRunnerOpts(t *testing.T) {
 		})
 	}
 }
+
+func TestContext(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	defer cancel()
+
+	p, err := syntax.Parse(strings.NewReader("while true; do true; done;"), "", 0)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	buff := bytes.NewBuffer(nil)
+	r := Runner{
+		File:    p,
+		Stdout:  buff,
+		Context: ctx,
+	}
+	endChan := make(chan struct{})
+
+	go func() {
+		if err := r.Run(); err != nil {
+			t.Fatal(err)
+		}
+		endChan <- struct{}{}
+	}()
+
+	select {
+	case <-time.After(time.Second * 6):
+		t.Error("Program was not killed in 6 seconds")
```

Comment:
If you make the timeout above much lower than `3s`, I assume this could easily be much lower like `1s`.